book/line_editor.md

```diff
@@ -1,124 +1,212 @@
+| <kbd>Shift</kbd>+<kbd>Enter</kbd>          | Insert newline                      |
+| <kbd>Alt</kbd>+<kbd>Enter</kbd>            | Insert newline                      |
+| <kbd>Backspace</kbd>                       | Backspace                           |
+| <kbd>End</kbd>                             | Move to end of line                 |
+| <kbd>End</kbd>                             | Complete history hint               |
+| <kbd>Home</kbd>                            | Move to line start                  |
+| <kbd>Ctrl</kbd>+<kbd>C</kbd>               | Cancel current line                 |
+| <kbd>Ctrl</kbd>+<kbd>L</kbd>               | Clear screen                        |
+| <kbd>Ctrl</kbd>+<kbd>R</kbd>               | Search history                      |
+| <kbd>Ctrl</kbd>+<kbd>→</kbd> (Right Arrow) | Complete history word               |
+| <kbd>Ctrl</kbd>+<kbd>→</kbd> (Right Arrow) | Move word right                     |
+| <kbd>Ctrl</kbd>+<kbd>←</kbd> (Left Arrow)  | Move word left                      |
+| <kbd>↑</kbd> (Up Arrow)                    | Move up                             |
+| <kbd>Ctrl</kbd>+<kbd>P</kbd>               | Move up                             |
+| <kbd>↑</kbd> (Up Arrow)                    | Move menu up                        |
+| <kbd>Ctrl</kbd>+<kbd>P</kbd>               | Move menu up                        |
+| <kbd>↓</kbd> (Down Arrow)                  | Move down                           |
+| <kbd>Ctrl</kbd>+<kbd>N</kbd>               | Move down                           |
+| <kbd>↓</kbd> (Down Arrow)                  | Move menu down                      |
+| <kbd>Ctrl</kbd>+<kbd>N</kbd>               | Move menu down                      |
+| <kbd>←</kbd> (Left Arrow)                  | Move left                           |
+| <kbd>Ctrl</kbd>+<kbd>B</kbd>               | Move left                           |
+| <kbd>←</kbd> (Left Arrow)                  | Move menu left                      |
+| <kbd>Ctrl</kbd>+<kbd>B</kbd>               | Move menu left                      |
+| <kbd>→</kbd> (Right Arrow)                 | Move right                          |
+| <kbd>Ctrl</kbd>+<kbd>F</kbd>               | Move right                          |
+| <kbd>→</kbd> (Right Arrow)                 | Move menu right                     |
+| <kbd>Ctrl</kbd>+<kbd>F</kbd>               | Move menu right                     |
+| <kbd>→</kbd> (Right Arrow)                 | History-hint complete               |
+| <kbd>Ctrl</kbd>+<kbd>F</kbd>               | History-hint complete               |
+| <kbd>Alt</kbd>+<kbd>F</kbd>                | History-hint complete one word      |
+| <kbd>Alt</kbd>+<kbd>←</kbd> (Left Arrow)   | History-hint complete one word less |
+
+### Vi-insert Keybindings
+
+| Key            | Event                    |
+| -------------- | ------------------------ |
+| <kbd>Esc</kbd> | Switch to Vi-normal mode |
```

Comment:
Maybe worth adding a sentence/entry that the above base bindings are available. Otherwise if you navigate to the particular section it may be weird.

Reply:
Thanks - added an intro-line to each section.